### DISCUSSION TITLE
Array intialization with one element problem

### DISCUSSION THREAD

**Discussion by lack-of-gravity-jack** [carbon-language/carbon-lang#3761] [Getting started]
I'm running on compiler explorer https://godbolt.org/ following this tutorial: https://www.youtube.com/watch?v=MMxbP8ME2Ag
`package helloworld api;`
`fn Main() -> i32 {`
`    var arr: [i32;1] = (3);`
`    return 0;`
`}`

I'm getting this error
`COMPILATION ERROR: <source>:7: type error in initializer of variable: 'i32' is not implicitly convertible to '[i32; 1]'
Compiler returned: 1`
If I add another element to the array and change the 1 to a 2, it works

`var arr: [i32;2] = (3,3); //works`
`var arr: auto = (3); //also works`

is this a problem with the compiler explorer?

**Comment by geoffromer** [comment]
The compiler explorer is correct. You can fix your example by writing `(3,)` instead of `(3)`. See [here](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/tuples.md#trailing-commas-and-single-element-tuples) for why that comma is necessary.
